### ISSUE TITLE
How do you find optimal parameters for mixup augmentation?

### ISSUE THREAD

**Issue by developer0hye** [ultralytics/yolov5#3380]
https://github.com/ultralytics/yolov5/blob/ba6f3f974bfc4a2968964dbe5eedea73c9f5efcb/utils/datasets.py#L536-L540

@glenn-jocher 

Hi, jocher!

How do you find optimal parameters for mixup augmentation?

`r = np.random.beta(8.0, 8.0)`

why are the values of alpha and beta  8?

**Comment by glenn-jocher** [comment]
@developer0hye this controls the ratio of mixing between img1 and img2. You can see the distribution of `r` here:
```python
import numpy as np
import matplotlib.pyplot as plt

x = np.random.beta(8.0, 8.0, 1000000)
plt.hist(x,500)
```
![Figure_1](https://user-images.githubusercontent.com/26833433/120067468-0a868d80-c07c-11eb-9d66-7c92086a7192.png)

The main issue discussing the mixup ratios and implementation for this repo is here:
https://github.com/ultralytics/yolov5/issues/357. The current values of 8.0 are used to implement a roughly 50/50 mixup with noise, i.e. maybe 0.2-0.8 3-sigma. I arrived at the 8.0 number after testing different betas, 0.5, 1.0, 2.0, 4.0, 8.0, and observed better results at higher betas.

**Comment by developer0hye** [comment]
@glenn-jocher 
Thanks for your reply!

**Comment by developer0hye** [comment]
@glenn-jocher 

Hi, jocher!

I recommend you do training with higher alpha and beta values like 32.0.

I got higher mAP with higher alpha and beta values on VOC dataset.

https://github.com/developer0hye/Simple-CenterNet

**Comment by glenn-jocher** [comment]
@developer0hye oh interesting! Higher beta values will tighten the mixing ratio close to 0.5. Do you have VOC mAP values vs beta?

**Comment by developer0hye** [comment]
@glenn-jocher 

It's my results.

alpha beta 8.0, mAP: 77.1
alpha beta 16.0 , mAP: 77.4
alpha beta 32.0, mAP: 77.6

**Comment by glenn-jocher** [comment]
@developer0hye oh awesome!! Yeah it seems I should increase the default beta then...
TODO: investigate increasing mixup betas.

**Comment by developer0hye** [comment]
@glenn-jocher 

I hope the graph of yolov5 be the below figure!

![image](https://user-images.githubusercontent.com/35001605/120485825-6db34f80-c3ef-11eb-99a5-6eca8ea55054.png)

**Comment by glenn-jocher** [comment]
@developer0hye haha, we'll add that graphic to the README on the next release.

**Comment by glenn-jocher** [comment]
@developer0hye we don't have much opportunity to test, but I'll assume your VOC results are accurate and generalize well to other datasets, so I've implemented this update in #3455. Thank you for your contributions!

**Comment by developer0hye** [comment]
@glenn-jocher 
Thanks! But... I'm a little worried that it might harm the performance of yolov5 on coco dataset.
